src/core/i18n/utils.js

```diff
@@ -230,6 +232,39 @@ function oneLineTranslationString(translationKey) {
   return translationKey;
 }
 
+type FormatFilesizeParams = {|
+  _filesize: typeof filesize,
+  i18n: I18nType,
+  size: number,
+|};
+
+// Translates a file size in bytes into a localized user-friendly format.
+export const formatFilesize = ({
+  _filesize = filesize,
+  i18n,
+  size,
+}: FormatFilesizeParams): string | null => {
+  const sizeStrings = {
+    // These are the expected values for the unit of measure returned by
+    // filesize. Realistically we shouldn't get anything back larger than TB.
+    B: i18n.gettext('%(formattedSize)s B'),
+    KB: i18n.gettext('%(formattedSize)s KB'),
+    MB: i18n.gettext('%(formattedSize)s MB'),
+    GB: i18n.gettext('%(formattedSize)s GB'),
+    TB: i18n.gettext('%(formattedSize)s TB'),
+  };
+
+  const [sizeNumber, sizeName] = _filesize(size).split(' ');
+  if (!sizeNumber || !sizeName) {
+    return null;
+  }
+
+  const formattedSize = i18n.formatNumber(sizeNumber);
+  const sizeString = sizeStrings[sizeName];
+
+  return sizeString ? i18n.sprintf(sizeString, { formattedSize }) : null;
```

Comment:
Similar to above, the case of an unmapped unit would be more helpful like this:

* Log an error so that Sentry will tell us that we have failed to map a known unit
* Return `i18n.formatNumber(sizeNumber)` without any units.